### PR TITLE
enable FERC explosion tags to be dimension specific

### PR DESCRIPTION
**Pull request by cmgosnell**
# PR Overview
working on [#2600](https://github.com/catalyst-cooperative/pudl/issues/2600)

# PR Checklist

- [x] Merge the most recent version of the branch you are merging into (probably `dev`).
- [ ] All CI checks are passing. [Run tests locally to debug failures](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#running-tests-with-tox)
- [x] Make sure you've included good docstrings.
- [ ] For major data coverage & analysis changes, [run data validation tests](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#data-validation)
- [ ] Include unit tests for new functions and classes.
- [ ] Defensive data quality/sanity checks in analyses & data processing functions.
- [ ] Update the [release notes](https://catalystcoop-pudl.readthedocs.io/en/latest/release_notes.html) and reference reference the PR and related issues.
- [x] Do your own explanatory review of the PR to help the reviewer understand what's going on and identify issues preemptively.

### CODE REVIEW

File: src/pudl/output/ferc1.py [hunk 1]
```diff
@@ -1630,8 +1653,10 @@ def single_valued_weights(cls, v: pd.DataFrame, values) -> pd.DataFrame:
             # That we are dealing with explicitly in building the trees below. Maybe it
             # should be happening here instead?
             logger.warning(
-                "Calculation forest nodes specified with conflicting weights:\n"
-                f"{v.loc[multi_valued_weights]}"
+                "Calculation forest nodes specified with conflicting weights."
+            )
+            logger.debug(
+                f"Nodes with conflicting weights:\n{v.loc[multi_valued_weights]}"
             )
```
Comment (cmgosnell) [inline src/pudl/output/ferc1.py:1660]: this was just reaaaly noisy.

File: src/pudl/package_data/ferc1/xbrl_factoid_rate_base_tags.csv [hunk 2]
```diff
@@ -1,218 +1,221 @@
...
+(less)_capital_stock_expense,(less)_capital_stock_expense,capital_stock_expense,balance_sheet_liabilities_ferc1,no,,,,
+(less)_discount_on_capital_stock,(less)_discount_on_capital_stock,discount_on_capital_stock,balance_sheet_liabilities_ferc1,no,,,,
+(less)_reaquired_capital_stock,(less)_reaquired_capital_stock,reacquired_capital_stock,balance_sheet_liabilities_ferc1,no,,,,
+_noncorporate_proprietorship_(non_major_only),_noncorporate_proprietorship_(non_major_only),noncorporate_proprietorship,balance_sheet_liabilities_ferc1,no,,,,
+appropriated_retained_earnings_amort_reserve_federal,appropriated_retained_earnings_amort_reserve_federal,appropriated_retained_earnings_amortization_reserve_federal,retained_earnings_ferc1,no,,,,
+depreciation,depreciation,accumulated_depreciation,electric_plant_depreciation_functional_ferc1,no,,future,,
+depreciation,depreciation,accumulated_depreciation,electric_plant_depreciation_functional_ferc1,yes,,in_service,,
+depreciation,depreciation,accumulated_depreciation,electric_plant_depreciation_functional_ferc1,yes,,leased,,
+depreciation,depreciation,accumulated_depreciation,electric_plant_depreciation_functional_ferc1,partial,,total,,
```
Comment (cmgosnell) [inline src/pudl/package_data/ferc1/xbrl_factoid_rate_base_tags.csv:221]: it is hard to see in the diff but i changed two things in this CSV:
- added dimension columns (almost entirely blank)
- added these four records where I know that the future `accumulated_depreciation` is not in rate base while the rest of the plant status are in rate base. I added `partial` for the total bc it is mixed.

File: src/pudl/output/ferc1.py [hunk 3]
```diff
@@ -912,10 +912,44 @@ class NodeId(NamedTuple):
...
+def _out_tags_ferc1(table_dimensions_ferc1) -> pd.DataFrame:
+    """Grab the stored table of tags and add infered dimension."""
+    # NOTE: there are a bunch of duplicate records in xbrl_factoid_rate_base_tags.csv
+    # Also, these tags are only applicable to the balance_sheet_assets_ferc1 table, but
+    # we need to pass in a dataframe with the right structure to all of the exploders,
+    # so we're just re-using this one for the moment.
+    tags_csv = (
+        importlib.resources.files("pudl.package_data.ferc1")
+        / "xbrl_factoid_rate_base_tags.csv"
+    )
```
Comment (cmgosnell) [inline src/pudl/output/ferc1.py:925]: made this tags table an asset bc that seemed more straightforward.

File: src/pudl/output/ferc1.py (hunk 1 above)
Comment (zaneselvans) [reply src/pudl/output/ferc1.py:1660]: I thought I shut this guy up on my branch as well...

**Review by zaneselvans — COMMENTED** [review]
This all seems remarkably straightforward!

I think the only issue is there are probably some conflicts between this and the branch you're merging into.

**Comment by zaneselvans** [comment]
I've resolved the tiny conflict and am going to investigate why you're getting sliiiightly more conflicting weights than expected (11.15% vs. 9.95%) and than seen on my branch upstream.

**Comment by zaneselvans** [comment]
When I run the explosions locally I still get the 9.95% so I suspect what's happening is there's a sliiiighly different number of facts actually showing up in the subset of data we see in CI vs. the full data, so I am going to go ahead and merrrrge.

**Review by zaneselvans — APPROVED** [review]

**Pull request merged**